### PR TITLE
[CALCITE-4972] Qualify names for repeated structs in getFieldOrigins

### PR DESCRIPTION
**Pull request by DeaconDesperado**
Addresses [CALCITE-4972](https://issues.apache.org/jira/browse/CALCITE-4972).

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java [hunk 1]
```diff
@@ -9210,6 +9210,23 @@ public void _testGroupExpressionEquivalenceParams() {
...
                 + " null,"
                 + " null}"));
+
+    sql("select e.empno from dept_nested, unnest(employees) as e")
+        .assertFieldOrigin(
+            is("{CATALOG.SALES.DEPT_NESTED.EMPLOYEES.EMPNO}"));
+
+    sql("select * from UNNEST(ARRAY['a', 'b'])")
+        .assertFieldOrigin(
+            is("{null}"));
```
Comment (zabetak) [inline core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:9220]: nit: rather short so keep in everything in the same line

File: core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java [hunk 2]
```diff
@@ -9210,6 +9210,23 @@ public void _testGroupExpressionEquivalenceParams() {
...
+
+    sql("select * from UNNEST(ARRAY['a', 'b'], ARRAY['d', 'e'])")
+        .assertFieldOrigin(
+            is("{null, "
+                + "null}"));
```
Comment (zabetak) [inline core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:9225]: nit: idem put in the same line

File: core/src/main/java/org/apache/calcite/sql/validate/UnnestNamespace.java [hunk 3]
```diff
@@ -64,6 +64,33 @@ class UnnestNamespace extends AbstractNamespace {
...
   }
 
+  /**
+   * Given a field name from SelectScope, find the column in this
+   * UnnestNamespace it originates from.
+   *
+   * @param queryFieldName Name of column
+   * @return A SqlQualified if subfield comes from this unnest, null if not found
+   */
+  public @Nullable SqlQualified getColumnUnnestedFrom(String queryFieldName) {
```
Comment (zabetak) [inline core/src/main/java/org/apache/calcite/sql/validate/UnnestNamespace.java:74]: Do we need to expose this as public? How about package private?

**Review by zabetak — COMMENTED** [review]
LGTM, minor comments and waiting for the tests.

**Comment by DeaconDesperado** [comment]
Thank you @zabetak for the review!  Checked in with changes.

**Pull request closed without merging**